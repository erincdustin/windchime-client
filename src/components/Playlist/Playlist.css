.container {
  position: relative;
  width: 100%;
  height: 100px;
  margin: 20px auto;
  text-align: center;
}

.iframe-box{
  position: absolute;
  top: 20px;
  right: 0;
  bottom: 0;
  z-index: 1;
  width: 100%;
  height: 80px;
}

.loading {
  font-style: italic;
  color: rgb(38, 104, 116);
  position: absolute;
  top: 20px;
  z-index: 0;
  width: 100%;
  height: 80px;
  background-color: rgb(232, 234, 235);
}