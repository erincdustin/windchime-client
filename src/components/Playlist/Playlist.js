import React from 'react';
import './Playlist.css';

function Playlist(props) {

  const url = `https://open.spotify.com/embed/user/spotify/playlist/${props.playlistId}`;
  
  return (
    <div className="container">
      <iframe className="iframe-box" title={props.playlistId} src={url} frameBorder="0" allowTransparency="true" ></iframe>
    </div>
  );
}

export default Playlist;