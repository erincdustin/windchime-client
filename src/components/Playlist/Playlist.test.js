import React from 'react';
import renderer from 'react-test-renderer';
import { library } from '@fortawesome/fontawesome-svg-core'
import { faWind } from '@fortawesome/free-solid-svg-icons'
import Playlist from './Playlist';

library.add(faWind)

describe(`Playlist`, () => {
  const props = {
    playlistId: '12345'
  }

  it('renders as expected', () => {
    const tree = renderer
      .create(<Playlist 
        playlistId={props.playlistId}
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
