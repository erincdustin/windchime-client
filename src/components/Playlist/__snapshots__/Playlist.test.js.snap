// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist renders as expected 1`] = `
<div
  className="container"
>
  <div
    className="loading"
  >
    Loading...
  </div>
  <iframe
    className="iframe-box"
    frameBorder="0"
    src="https://open.spotify.com/embed/user/spotify/playlist/12345"
    title="12345"
  />
</div>
`;
